File: php-src/Uploader/Translations.php
<?php

namespace kalanis\UploadPerPartes\Uploader;


use kalanis\UploadPerPartes\Interfaces\IUppTranslations;


/**
 * Class Translations
 * @package kalanis\UploadPerPartes
 * Return translated quotes from backend
 * - necessary due many translation systems through web
 * For work extends this class and pass extension into your project
 */
class Translations implements IUppTranslations
{
    public function uppBadResponse(string $responseType): string
    {
        return 'Selected bad response type.';
    }

    public function uppTargetNotSet(): string
    {
        return 'The target is not set.';
    }

    public function uppTargetIsWrong(string $url): string
    {
        return sprintf('The target is set in a wrong way. Cannot determine it. *%s*', $url);
    }

    public function uppChecksumVariantIsWrong(string $variant): string
    {
        return sprintf('The checksum is set in a wrong way. Cannot determine it. *%s*', $variant);
    }

    public function uppDecoderVariantIsWrong(string $variant): string
    {
        return sprintf('The decoder is set in a wrong way. Cannot determine it. *%s*', $variant);
    }

    public function uppIncomingDataCannotDecode(): string
    {
        return 'Cannot decode incoming data!';
    }

    public function uppSentNameIsEmpty(): string
    {
        return 'Sent file name is empty.';
    }

    public function uppChecksumIsEmpty(): string
    {
        return 'There is no data for checksum on storage.';
    }

    public function uppDataEncoderVariantNotSet(): string
    {
        return 'The driving data encoder variant is not set!';
    }

    public function uppDataEncoderVariantIsWrong(string $variant): string
    {
        return sprintf('The driving data encoder is set in a wrong way. Cannot determine it. *%s*', $variant);
    }

    public function uppDataModifierVariantNotSet(): string
    {
        return 'The driving data modifier variant is not set!';
    }

    public function uppDataModifierVariantIsWrong(string $variant): string
    {
        return sprintf('The driving data modifier is set in a wrong way. Cannot determine it. *%s*', $variant);
    }

    public function uppKeyEncoderVariantNotSet(): string
    {
        return 'The driving data key encoder variant is not set!';
    }

    public function uppKeyEncoderVariantIsWrong(string $className): string
    {
        return sprintf('The driving data key encoder variant is set in a wrong way. Cannot determine it. *%s*', $className);
    }

    public function uppKeyModifierVariantNotSet(): string
    {
        return 'The driving data key modifier variant is not set!';
    }

    public function uppKeyModifierVariantIsWrong(string $className): string
    {
        return sprintf('The driving data key modifier variant is set in a wrong way. Cannot determine it. *%s*', $className);
    }

    public function uppDriveFileStorageNotSet(): string
    {
        return 'The driving file storage is not set correctly!';
    }

    public function uppDriveFileCannotRead(string $key): string
    {
        return sprintf('Cannot read *%s* driving file from its storage.', $key);
    }

    public function uppDriveFileCannotWrite(string $key): string
    {
        return sprintf('Cannot write *%s* driving file into its storage.', $key);
    }

    public function uppDriveFileAlreadyExists(string $driveFile): string
    {
        return sprintf('The driving file *%s* already exists in storage.', $driveFile);
    }

    public function uppTempEncoderVariantNotSet(): string
    {
        return 'The temporary storage encoder variant is not set!';
    }

    public function uppTempEncoderVariantIsWrong(string $variant): string
    {
        return sprintf('The temporary storage encoder variant is set in a wrong way. Cannot determine it. *%s*', $variant);
    }

    public function uppTempStorageNotSet(): string
    {
        return 'The temporary storage is not set correctly!';
    }

    public function uppFinalEncoderVariantNotSet(): string
    {
        return 'The final storage encoder variant is not set!';
    }

    public function uppFinalEncoderVariantIsWrong(string $variant): string
    {
        return sprintf('The final storage encoder variant is set in a wrong way. Cannot determine it. *%s*', $variant);
    }

    public function uppFinalStorageNotSet(): string
    {
        return 'The final storage is not set correctly!';
    }

    public function uppCannotReadFile(string $location): string
    {
        return sprintf('Cannot read file *%s*', $location);
    }

    public function uppCannotWriteFile(string $location): string
    {
        return sprintf('Cannot write file *%s*', $location);
    }

    public function uppDriveFileCannotRemove(string $key): string
    {
        return sprintf('Cannot remove drive file for upload *%s*', $key);
    }

    public function uppCannotRemoveData(string $location): string
    {
        return sprintf('Cannot remove drive file for upload *%s*', $location);
    }

    public function uppCannotTruncateFile(string $location): string
    {
        return sprintf('Cannot truncate file *%s*', $location);
    }

    public function uppSegmentOutOfBounds(int $segment): string
    {
        return sprintf('Segment *%d* is out-of-bounds.', $segment);
    }
}

File: kw_upload/uploader/translations.py

class Translations:
    """
     * Return translated quotes from backend
     * - necessary due many translation systems through web
     * For work extends this class and pass that extension into your project
    """
    def upp_sent_name_is_empty(self) -> str:
        return 'SENT FILE NAME IS EMPTY'

    def upp_upload_name_is_empty(self) -> str:
        return 'UPLOAD FILE NAME IS EMPTY'

    def upp_shared_key_is_empty(self) -> str:
        return 'SHARED KEY IS EMPTY'

    def upp_shared_key_is_invalid(self) -> str:
        return 'SHARED KEY IS INVALID'

    def upp_key_variant_not_set(self) -> str:
        return 'KEY VARIANT NOT SET'

    def upp_key_variant_is_wrong(self, class_name: str) -> str:
        return 'KEY VARIANT IS WRONG'

    def upp_target_dir_is_empty(self) -> str:
        return 'TARGET DIR IS NOT SET'

    def upp_drive_file_already_exists(self, drive_file: str) -> str:
        return 'DRIVEFILE ALREADY EXISTS'

    def upp_drive_file_not_continuous(self, drive_file: str) -> str:
        return 'DRIVEFILE IS NOT CONTINUOUS'

    def upp_drive_file_cannot_remove(self, key: str) -> str:
        return 'DRIVEFILE CANNOT BE REMOVED'

    def upp_drive_file_variant_not_set(self) -> str:
        return 'DRIVEFILE VARIANT NOT SET'

    def upp_drive_file_variant_is_wrong(self, class_name: str) -> str:
        return 'DRIVEFILE VARIANT IS WRONG'

    def upp_drive_file_cannot_read(self, key: str) -> str:
        return 'CANNOT READ DRIVEFILE'

    def upp_drive_file_cannot_write(self, key: str) -> str:
        return 'CANNOT WRITE DRIVEFILE'

    def upp_cannot_remove_data(self, location: str) -> str:
        return 'CANNOT REMOVE DATA'

    def upp_read_too_early(self, key: str) -> str:
        return 'READ TOO EARLY'

    def upp_cannot_open_file(self, location: str) -> str:
        return 'CANNOT OPEN FILE'

    def upp_cannot_read_file(self, location: str) -> str:
        return 'CANNOT READ FILE'

    def upp_cannot_seek_file(self, location: str) -> str:
        return 'CANNOT SEEK FILE'

    def upp_cannot_write_file(self, location: str) -> str:
        return 'CANNOT WRITE FILE'

    def upp_cannot_truncate_file(self, location: str) -> str:
        return 'FILE CANNOT TRUNCATE'

    def upp_segment_out_of_bounds(self, segment: int) -> str:
        return 'SEGMENT OUT OF BOUNDS'

    def upp_segment_not_uploaded_yet(self, segment: int) -> str:
        return 'SEGMENT NOT UPLOADED YET'
